zpay32/invoice.go

```diff
@@ -271,37 +274,47 @@ func Decode(invoice string) (*Invoice, error) {
 	if err != nil {
 		return nil, err
 	}
+	// First two characters of HRP should be "ln".
+	if hrp[:2] != "ln" {
+		return nil, fmt.Errorf("prefix should be \"ln\"")
+	}
 
 	// We expect the human-readable part to at least have ln + two chars
 	// encoding the network.
 	if len(hrp) < 4 {
```

Comment:
The check was in the right place, it just needed to be added back. We can't guarantee the `hrp` will have a length large enough to see if it contains "ln", so we should check its length first.

Reply:
Ah thanks for catching this, I'll put it back in the right place.